### PR TITLE
[formrecognizer] some edits for code gen

### PR DESCRIPTION
**Pull request by kristapratico**

### CODE REVIEW

File: specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.0/FormRecognizer.json [hunk 1]
```diff
@@ -490,6 +453,100 @@
...
+        "summary": "List Custom Models",
+        "description": "Get information about all custom models",
+        "operationId": "ListCustomModels",
+        "consumes": [],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "op",
```
Comment (lmazuel) [inline specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.0/FormRecognizer.json]: Should be required with a single enum value "full":
https://github.com/Azure/azure-rest-api-specs/blob/afe2cdb70a548ebf0348e433bf7d0437a2a8c986/specification/storage/data-plane/Microsoft.BlobStorage/preview/2019-12-12/blob.json#L161-L168

File: specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.0/FormRecognizer.json [hunk 2]
```diff
@@ -490,6 +453,100 @@
...
+        "summary": "Get Custom Models",
+        "description": "Get information about all custom models",
+        "operationId": "GetCustomModels",
+        "consumes": [],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "op",
```
Comment (lmazuel) [inline specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.0/FormRecognizer.json]: Should be required with a single enum value "summary":
https://github.com/Azure/azure-rest-api-specs/blob/afe2cdb70a548ebf0348e433bf7d0437a2a8c986/specification/storage/data-plane/Microsoft.BlobStorage/preview/2019-12-12/blob.json#L161-L168

File: specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.0/FormRecognizer.json [hunk 3]
```diff
@@ -493,6 +453,98 @@
...
+            "in": "query",
+            "description": "Specify whether to return summary or full list of models.",
+            "required": true,
+            "default": "full",
```
Comment (lmazuel) [inline specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.0/FormRecognizer.json]: Please remove the default

File: specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.0/FormRecognizer.json [hunk 4]
```diff
@@ -493,6 +453,98 @@
...
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "op",
+            "in": "query",
+            "description": "Specify whether to return summary or full list of models.",
+            "required": true,
+            "default": "summary",
```
Comment (lmazuel) [inline specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.0/FormRecognizer.json]: Please remove the default

File: specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.0/FormRecognizer.json [hunk 5]
```diff
@@ -493,6 +453,96 @@
       }
     }
   },
+  "x-ms-paths": {
+    "/custom/models?op=full": {
```
Comment (johanste) [inline specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.0/FormRecognizer.json]: I'd use a different "fake"/uniquefying parameter - e.g. `?type=custom`

Comment (kristapratico) [reply specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.0/FormRecognizer.json]: Maybe I'm misunderstanding, but `?op=full` is the name/value for the enum. Or does that not matter here?

Comment (lmazuel) [reply specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.0/FormRecognizer.json]: In `x-ms-paths`, the part behind the ? could be anything, and has no particular value. Could be (literally) `/custom/models?This is an identifier I want it long`

That being said, I think it's a good practice to keep it to the semantic of the call, and here it's about doing `op=full`, so I would keep it with that name

**Review by lmazuel — APPROVED** [review]

**Pull request merged**